datajoint/fetch.py

```diff
@@ -93,71 +107,104 @@ def order_by(self, *args):
     @property
     def as_dict(self):
         """
+        DEPRECATED
+
         Changes the state of the fetch object to return dictionaries.
         :return: a copy of the fetch object
         Example:
 
         >>> my_relation.fetch.as_dict()
 
         """
+        warnings.warn('Use of `as_dict` on `fetch` object is deprecated. Please use `as_dict` keyword arguments in the '
+                      'call to `fetch`/`keys` instead', stacklevel=2)
         ret = Fetch(self)
         ret.sql_behavior['as_dict'] = True
         return ret
 
     def limit(self, limit):
         """
+        DEPRECATED
+
         Limits the number of items fetched.
 
         :param limit: limit on the number of items
         :return: a copy of the fetch object
         """
+        warnings.warn('Use of `limit` on `fetch` object is deprecated. Please use `limit` keyword arguments in '
+                      'the call to `fetch`/`keys` instead', stacklevel=2)
         ret = Fetch(self)
         ret.sql_behavior['limit'] = limit
         return ret
 
     def offset(self, offset):
         """
+        DEPRECATED
+
         Offsets the number of itms fetched. Needs to be applied with limit.
 
         :param offset: offset
         :return: a copy of the fetch object
         """
+
+        warnings.warn('Use of `offset` on `fetch` object is deprecated. Please use `offset` keyword arguments in '
+                      'the call to `fetch`/`keys` instead', stacklevel=2)
         ret = Fetch(self)
         if ret.sql_behavior['limit'] is None:
             warnings.warn('You should supply a limit together with an offset,')
         ret.sql_behavior['offset'] = offset
         return ret
 
-    def __call__(self, **kwargs):
+    def __call__(self, *attrs, **kwargs):
         """
         Fetches the relation from the database table into an np.array and unpacks blob attributes.
 
+        :param attrs: OPTIONAL. one or more attributes to fetch. If not provided, the call will return
+        all attributes of this relation. If provided, returns tuples with an entry for each attribute.
         :param offset: the number of tuples to skip in the returned result
         :param limit: the maximum number of tuples to return
         :param order_by: the list of attributes to order the results. No ordering should be assumed if order_by=None.
         :param as_dict: returns a list of dictionaries instead of a record array
         :return: the contents of the relation in the form of a structured numpy.array
         """
+        # if 'order_by' passed in a string, make into list
+        if isinstance(kwargs.get('order_by'), str):
+            kwargs['order_by'] = [kwargs['order_by']]
+
         sql_behavior = update_dict(self.sql_behavior, kwargs)
         ext_behavior = update_dict(self.ext_behavior, kwargs)
+        total_behavior = dict(sql_behavior)
```

Comment:
Yes - just note that we are still going to keep backward compatible behavior for a few rounds and hence I'm not drastically changing the internals yet.